### ISSUE TITLE
Ruby System Shell Commands have empty string responses on Windows when not launching SketchUp from command line

### ISSUE THREAD

**Issue by kengey** [SketchUp/api-issue-tracker#216]
When developing plugins I always launch sketchup from command line. Very recently a client was unable to use my extension because of a part that executes a shell command and expecting return from it was not functioning. It seems that, if you launch SketchUp by any other way than from command line (thus, double clicking the executable, or a shortcut to that executable or...) make ruby subshell commands always return an empty string.
I have tested and found the same behavior on this with all version installed on my machine:

- 17.2.2555 64-bit
- 18.0.16975 64-bit
- 19.0.685 64-bit

I have uploaded a youtube video:
[https://youtu.be/G_nveWVCo18](https://youtu.be/G_nveWVCo18
)

**Comment by thomthom** [comment]
This sounds similar to some other issues we've had in the past where stdlib haven't loaded properly...

@BugraBarin  - do you recall what other issues we had when SU was launched in different ways?

**Comment by BugraBarin** [comment]
Yes this has been a long-standing issue. We even debugged into the Ruby source code to understand what's going on and we saw that the pipe between SketchUp and the subprocess kept breaking for an unknown reason. (Windows error code was literally unknown). We stopped chasing this at that point. So executing a shell process from Ruby is currently unsupported. Sorry :/

**Comment by DanRathbun** [comment]
But this is MS Windows platform specific, correct ?

**Comment by taustin73** [comment]
SU-44316

**Comment by DanRathbun** [comment]
BTW, the workaround for this issue on Windows is to redirect command output to a file and then have SketchUp Ruby read the file.
```ruby
output_file = File.join(output_path, 'dir.txt')
Dir.chdir(some_path) do
  %x[dir > #{output_file}]
end
sleep 0.5
if File.exist?(output_file)
  text = File.read(output_file)
end
```
\* Yes, I know the [`Dir`](https://ruby-doc.org/core-2.7.2/Dir.html) class can easily read filenames from directories. It's just an example command.

**Comment by kalpana-ghodeswar** [comment]
Sorry, we have attempted few times in the past but not able to fix it.
